rewatch/src/build.rs

```diff
@@ -65,6 +66,24 @@ pub fn get_compiler_args(path: &Path, build_dev_deps: bool) -> Result<String> {
         packages::read_config(&workspace_root.to_owned().unwrap_or(package_root.to_owned()))?;
     let rescript_config = packages::read_config(&package_root)?;
 
+    // We need to figure out if the source file is listed as "type":"dev"
+    // To do this, we can construct the Package information for the nearest config.
+    let package = packages::make_package(root_rescript_config.clone(), &package_root, false, true);
+    let mut packages_map: AHashMap<String, Package> = AHashMap::new();
+    let package_name = root_rescript_config.name.clone();
+    packages_map.insert(package_name.clone(), package);
+    // We need to populate the source_files meta data.
+    // TODO: filter with current file
+    let packages_map = packages::extend_with_children(
```

Comment:
This is very expensive, this would scan all the files in the package. I think it would be better to see if the current file matches any of the dev paths.